Makefile

```diff
@@ -1,7 +1,7 @@
 export CARGO_PROFILE_RELEASE_CODEGEN_UNITS = 1
 export CARGO_PROFILE_RELEASE_LTO = fat
 export DOCKER_BUILDKIT = 1
-export RUSTFLAGS = -D warnings
+export RUSTFLAGS = "-D warnings -C target-cpu=native"
```

Comment:
We cannot do this. The binary built by the `Makefile` should be able to run on any machine that satisfies the minimum requirements, which are going to be significantly lower than those available on a typical build machine.

Reply:
It seems `rust-rocksdb` 0.16 have unconditionally enabled sse2/sse4.1/sse4.2 now: https://github.com/rust-rocksdb/rust-rocksdb/blob/v0.16.0/librocksdb-sys/build.rs#L106.
Let me look for a workaround.

Reply:
What about enabling only sse2/sse4.1/sse4.2 since they're in fact already a requirement (implicit) for nodes?

Reply:
I originally suggested doing something like that in https://github.com/near/nearcore/issues/5487. I personally think it makes sense to enable at least some of these features unconditionally, but others might disagree.

Reply:
I see. Yeah that's cool. Sorry I didn't understand how it's related first time you shared the PR. How do we sync these 2 PRs? Should I wait for you or I can go ahead?

Reply:
> I see. Yeah that's cool. Sorry I didn't understand how it's related first time you shared the PR. How do we sync these 2 PRs? Should I wait for you or I can go ahead?

To me it seems like we should have three separate PRs:
1. Enable SSE et al in build flags.
2. Add CPU features checking to release builds.
3. Upgrade RocksDB.

Currently we have two PRs. One which covers 1 and 2 and another which covers 1 and 3.

Reply:
> I see. Yeah that's cool. Sorry I didn't understand how it's related first time you shared the PR. How do we sync these 2 PRs? Should I wait for you or I can go ahead?

Feel free to land your changes first; I don't see myself coming back to my own PR anytime soon.

---

> Add CPU features checking to release builds.

This is already a thing, my PR just refactors it, but lets discuss that particular detail over there rather than here.

> To me it seems like we should have three separate PRs:

On one hand I'm inclined to agree. More granluar changesets would avoid blocking part of the PR that seems good based on concerns with a different part of the PR. Smaller changesets are easier to review as well. On another hand, there are also a reason to keep 1 and 3 together as well – there isn't all that much of a reason to make the 1st change in isolation if we end up never making the 3rd one.

Reply:
I'm okay with splitting this to 2 commits. Please let me know if it's a blocker.

On the other hand I agree that those should go together: hypothetically if later on we decide to revert flags change it can lead to performance degradation if we don't revert 0.16 -> 0.18 as well.